compiler/damlc/tests/src/DA/Test/DataDependencies.hs

```diff
@@ -1651,6 +1651,22 @@ tests tools@Tools{damlc,validate,oldProjDar} = testGroup "Data Dependencies" $
         , "z f g = f <<< g &&& id"
         ]
 
+    , simpleImportTestOptions "No 'inaccessible RHS' when pattern matching on interface"
+        [ "--target=1.dev" ]
+        [ "module Lib where"
+
+        , "interface I"
+        ]
+        [ "{-# OPTIONS_GHC -Werror #-}"
+        , "module Main where"
+        , "import Lib"
+
+        , "isJustI : Optional I -> Bool"
+        , "isJustI mI = case mI of"
+        , "  Some _ -> True"
```

Comment:
side note do you understand why `isJustI : I -> I` is not sufficient here? I thought we enable `-XStrict` and the strict pattern match should make it fail with the same warning but it doesn’t in my tests.

Reply:
As far as I could tell, all versions of GHC accept `idV` and `idV'`

```haskell
{-# LANGUAGE Strict #-}
{-# OPTIONS_GHC -Wall -Werror #-}

data Void

idV :: Void -> Void
idV v = v

idV' :: Void -> Void
idV' v = case v of w -> w
```

it seems that the pattern match checks only kick in for pattern matches other than `VarPat`, since those are treated as plain `let`s (see [GHC.HsToCore.Pmc.Desugar#L111](https://gitlab.haskell.org/ghc/ghc/-/blob/master/compiler/GHC/HsToCore/Pmc/Desugar.hs#L111)). 

Considering the documentation for `-XStrict`,

> * Function definitions
>
>     When the user writes
>
>     ```
>     f x = ...
>     ```
>
>     we interpret it as if they had written
> 
>     ```
>     f !x = ...
>     ```
> -- [Glasgow Haskell Compiler 9.2.2 User's Guide - 6.14. Bang patterns and Strict Haskell - 6.14.3. Strict-by-default pattern bindings](https://downloads.haskell.org/ghc/latest/docs/html/users_guide/exts/strict.html#strict-by-default-pattern-bindings)

this looks like a bug in the interaction between the pattern match checker and the `Strict` pragma

Reply:
thanks!